spyder/plugins/editor/widgets/codeeditor.py

```diff
@@ -1129,10 +1129,13 @@ def process_signatures(self, params):
 
                 parameter_idx = signature_params['activeParameter']
                 parameters = signature_data['parameters']
-                parameter_data = parameters[parameter_idx]
+                parameter = None
+
+                if len(parameters) > 0:
+                    parameter_data = parameters[parameter_idx]

```

Comment:
I think that this line is related to #10372.
Sometimes the idx is greater than the length when using kite.